File: 8kyuCountingSheep.js
var array1 = [true,  true,  true,  false,
    true,  true,  true,  true ,
    true,  false, true,  false,
    true,  false, false, true ,
    true,  true,  true,  true ,
    false, false, true,  true ];

function countSheeps(array1){
    let result = array1.filter(e => e === true);
    return result.length
}

//test countSheeps(array1)

File: C#/8kyu/8kyuCountingSheep.cs
using System.Linq;

public static class Kata
{
  public static int CountSheeps(bool[] sheeps)
  {
    int result =0;
    for (int i = 0; i< sheeps.Length; i++) {
      if (sheeps[i] == true) {
        result++;
      }
    }
    return result;
  }
}